### PR TITLE
Feat/gzip

### PR DESCRIPTION
**Pull request by zflamig**

### CODE REVIEW

File: kube/services/revproxy/00nginx-config.yaml [hunk 1]
```diff
@@ -139,7 +139,15 @@ data:
...
-        gzip_types application/javascript;
+        gzip_proxied any;
+        gzip_types
+          text/css
+          text/javascript
+          text/xml
+          text/plain
+          application/javascript
+          application/x-javascript
+          application/json;
```
Comment (frickjack) [inline kube/services/revproxy/00nginx-config.yaml]: I think we decided not to gzip json due to the BREACH attack:
   https://en.wikipedia.org/wiki/BREACH

File: kube/services/netpolicy/networkpolicy_fence_templ.yaml [hunk 2]
```diff
@@ -23,6 +23,9 @@ spec:
       - podSelector:
           matchLabels:
              app: revproxy
+      - podSelector:
```
Comment (frickjack) [inline kube/services/netpolicy/networkpolicy_fence_templ.yaml]: Why does shiny need to communicate directly with the fence service?
Would it be better if client code accessed our services via our standard public endpoint, than have one way to do things for workspace clients like shiny and another for external clients ?

File: kube/services/revproxy/00nginx-config.yaml (hunk 1 above)
Comment (zflamig) [reply kube/services/revproxy/00nginx-config.yaml]: I don't believe we actually reflect user data anywhere so we shouldn't be vulnerable to this?

Comment (zflamig) [reply kube/services/revproxy/00nginx-config.yaml]: Actually thats not true, we reflect names in graphql queries so you could potentially do it there. Disabling for the graphql endpoint

**Pull request merged**